File: php/features.php
<?php
declare(strict_types=1);

// UseragentParser SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UseragentParserFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UseragentParserBaseFeature();
            case "test":
                return new UseragentParserTestFeature();
            default:
                return new UseragentParserBaseFeature();
        }
    }
}

File: py/features.py
# UseragentParser SDK feature factory

from feature.base_feature import UseragentParserBaseFeature
from feature.test_feature import UseragentParserTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UseragentParserBaseFeature(),
        "test": lambda: UseragentParserTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
